api/v1alpha3/azuremachine_validation.go

```diff
@@ -51,3 +51,20 @@ func ValidateUserAssignedIdentity(identityType VMIdentity, userAssignedIdentetie
 
 	return allErrs
 }
+
+// ValidateOSDisk validates the OSDisk spec
+func ValidateOSDisk(osDisk OSDisk, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if osDisk.DiskSizeGB <= 0 {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("DiskSizeGB"), "", "Disk Size cannot be zero or negative"))
```

Comment:
This also needs to be smaller or equal to 2048: https://docs.microsoft.com/en-us/azure/virtual-machines/windows/managed-disks-overview#os-disk

Reply:
We should also not require the disk size to be set, Azure APIs do default the disk size if it's empty, but that's out of scope of this PR since it would require changing the type which would be a breaking change.

https://github.com/Azure/azure-sdk-for-go/blob/master/services/compute/mgmt/2019-12-01/compute/models.go#L10794

Reply:
@CecileRobertMichon Does it make sense to add a validation of the actual values of the `StorageAccountType` based on the [allowed SKU Names](https://docs.microsoft.com/en-us/powershell/module/az.compute/new-azdiskconfig?view=azps-1.8.0) and `OSType` based on the accepted values of `OsType` (Windows/Linux)?

Reply:
CAPZ doesn't support Windows nodes at the moment so I think only `Linux` should be valid for now. If we allow `Windows` it might be confusing for users and lead them to think that it is supported, so for now I would vote for only allowing `Linux`.

I think it makes sense to validate that `StorageAccountType` is a valid SKU name.